File: bubble_sort.py
# 1.8358230590820312e-05 for 10
# 0.00047230720520019530 for 100
# 0.05660581588745117000 for 1000
# 6.01831507682800300000 for 10000
def bubble_sort(values):
    for i in range(0, len(values)):
        for j in range(1, len(values) - i):
            if values[j - 1] > values[j]:
                values[j - 1], values[j] = values[j], values[j - 1]

File: BubbleSort.java
// 10: 3.58E-6
// 100: 1.46964E-4
// 1000: 0.00325546
// 10000: 0.06482598
// 100000: 2.542452682
public class BubbleSort {
    public static void sort(double[] values){
        for(int i = 0;i < values.length;i++){
            for(int j = 0;j < values.length - i - 1;j++){
                if(values[j] > values[j + 1]){
                    double tmp = values[j];
                    values[j] = values[j + 1];
                    values[j + 1] = tmp;
                }
            }
        }
    }

    public static void test(int length){
        long start = System.nanoTime();
        double[] values = new double[length];
        for(int i = 0;i < length;i++){
            values[i] = length - i;
        }
        sort(values);
        long performance = System.nanoTime() - start;
        for(double value : values){
            System.out.println(value);
        }
        System.out.println(performance / 1000000000.0);
    }
}
